21-iteration.Rmd

```diff
@@ -610,94 +610,93 @@ Si estás familiarizado con la familia de funciones apply en R base, podrás hab
+Cuando la función es exitosa, el elemento `result` contiene el resultado y el elemento `error` es `NULL`. Cuando la función falla, el elemento `result` es `NULL` y el elemento `error` contiene un objeto de error.
 
-`safely()` is designed to work with map:
+`safely()` está diseñada para funcionar con map:
 
 ```{r}
 x <- list(1, 10, "a")
 y <- x %>% map(safely(log))
 str(y)
 ```
 
-This would be easier to work with if we had two lists: one of all the errors and one of all the output. That's easy to get with `purrr::transpose()`:
+Esto sería más fácil de trabajar si tuvieramos dos listas: una con todos los errores y otra con todas las salidas, Esto es fácil de obtener con `purrr::transpose()`:
 
 ```{r}
 y <- y %>% transpose()
 str(y)
 ```
 
-It's up to you how to deal with the errors, but typically you'll either look at the values of `x` where `y` is an error, or work with the values of `y` that are ok:
+Queda a tu criterio como manejar los errores, pero típicamente puedes mirar los valores de `x` donde `y` es un error, o trabajar con los valores que `y` que están ok:
 
 ```{r}
 is_ok <- y$error %>% map_lgl(is_null)
 x[!is_ok]
 y$result[is_ok] %>% flatten_dbl()
 ```
 
-Purrr provides two other useful adverbs:
+Purrr provee otros dos adverbios de cantidad:
+
+*   Como `safely()`, `possibly()` siempre da éxito. Es más simple que `safely()`, 
+    ya que le das un valor por defecto para devolver cuando hay un error.
 
-*   Like `safely()`, `possibly()` always succeeds. It's simpler than `safely()`, 
-    because you give it a default value to return when there is an error. 
-    
     ```{r}
     x <- list(1, 10, "a")
     x %>% map_dbl(possibly(log, NA_real_))
     ```
     
-*   `quietly()` performs a similar role to `safely()`, but instead of capturing
-    errors, it captures printed output, messages, and warnings:
-    
+*   `quietly()` tiene un rol similar a `safely()`, pero en lugar de capturar
+    los errores, captura la salida impresa, mensajes y advertencias:
+
     ```{r}
     x <- list(1, -1)
     x %>% map(quietly(log)) %>% str()
     ```
 
-## Mapping over multiple arguments
+## Mapear sobre múltiples argumentos
 
-So far we've mapped along a single input. But often you have multiple related inputs that you need iterate along in parallel. That's the job of the `map2()` and `pmap()` functions. For example, imagine you want to simulate some random normals with different means. You know how to do that with `map()`:
+Hasta ahora hemos mapeado sobre una única entrada. Pero a menudo tienes multiples entradas relacionadas y necesitas iterar sobre ellas en paralelo. Ese es el trabajo de las funciones `map2()` and `pmap()` functions. Por ejemplo, imagina que debes simular normales aleatorias con distintas medias. Ya sabes hacerlo con `map()`:
```

Comment:
Ese es el trabajo de las funciones `map2()` and `pmap()` functions.
Eliminar el "functions" al final de la oracion.